numba/targets/base.py

```diff
@@ -667,6 +713,29 @@ def cast(self, builder, val, fromty, toty):
         elif fromty == types.none and isinstance(toty, types.Optional):
             return self.make_optional_none(builder, toty.type)
 
+        elif (isinstance(toty, types.Optional) and
```

Comment:
In light of the various Optional enhancements, it would be nice to add tests to `test_optional`, IMHO.